lib/routes.js

```diff
@@ -0,0 +1,34 @@
+const axios = require('axios');
+
+module.exports = function (self, options) {
+  self.addRoutes = function() {
+    self.route('post', 'page-scan-modal', async (req, res) => {
+      if (!self.apos.permissions.can(req, 'admin')) {
+        return res.status(403).send('forbidden');
+      }
+
+      if (!req.body || !req.body.href) {
+        return res.status(400).send('bad request');
+      }
+
+      try {
+        const { href } = req.body;
+
+        const { data } = await axios.get(href, {
```

Comment:
You must verify that this URL belongs to the site. Otherwise the login session cookie could be transmitted inappropriately to another site.

Reply:
`req.absoluteUrl` should be helpful in figuring that out.

Reply:
Oh my bad, Actually we request internal links from the client side, and only external links from this route. So we don't have to passe cookie at all here.
Thanks for noticing it

Reply:
Right, I used `baseUrl` to check we're on the same domain !